src/Components/Button/ButtonComponent.tsx

```diff
@@ -13,27 +13,42 @@ const ButtonComponent = ({
   disabled,
   onClick,
 }: ButtonProps) => {
-  const [isPressed, setIsPressed] = useState(false);
-
   const handlePress = (event: React.MouseEvent<HTMLButtonElement>) => {
-    setIsPressed((pervState) => !pervState);
     onClick(event);
```

Comment:
if you dont need the setIsPressed to fire  in the `handlePress` handler, you dont need this handler at all.
you can just do 
```
    <Button
      sx={getButtonStyles(type, size)}
      onClick={onClick}
      disabled={disabled}
    >
``` 
in line 21